src/framework.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * File: framework.js 
+ */
+var _ = require('underscore');
+var util = require('util');
+
+var stateFactory = require('./create-state');
+var vnis = require('./vnis');
+
+/**
+ * This module implements the core of the RDF Pipeline Framework.  
+ */
+module.exports = {
+
+    componentFactory: function( nodeDef, wrapper ) { 
+        
+        var compFactory = require('./noflo-component-factory');
+        return compFactory(
+                _.defaults( {
+                     // Add an ondata function to be invoked every time we receive new data
+                     inPorts: _.mapObject(nodeDef.inPorts, function( port, portName ) {  
+                         return _.extend({
+                             ondata: function(payload, socketIndex) {
+                                   var that = this; 
```

Comment:
JS noobie question:  Is the line above actually needed in this case?  Since no function is being returned in the next line, could the next line have been written as follows instead?
<pre>
return ondata.call( this, payload, portName, socketIndex, port );
</pre>